docs/source/fitfunctions/FunctionQDepends.rst

```diff
@@ -0,0 +1,30 @@
+.. _func-FunctionQDepends:
+
+================
+FunctionQDepends
+================
+
+.. index:: FunctionQDepends
+
+Description
+-----------
+
+This fitting function is the base class for all fitting functions that have:
+
+- A range of energy transfers as their one-dimensional domain from which to take values
+- The magnitude of the momemtum transfer, :math:`Q`, as one of their attributes.
+
+Fitting functions for QENS data depending on :math:`Q` should derive from this class.
+
+There are two ways to update attribute :math:`Q` in the fit funtion:
+
+- The user inputs a particular value.
+- The spectrum contains :math:`Q` in addition to the range of energy transfers.
+
+Conflict may arise when both options are available. In that case, priority is given to the :math:`Q`-value contained
```

Comment:
There are two mutually exclusive user cases and I had to give preference to one of them. The first user case is the one you described, the case when you have corrupt input data. The second user case is when you have correct data but the inexperienced user screws up by manually overriding the Q-value, typically fumbling with the Q-attribute in one of the Mantid GUIs.
In my experience, the latter user case is much more common because most of our users are inexperienced and because reduction will produce spectra with correct Q-values. In the rare event of corrupt data I rather have the user correct their data _outside_ of the fitting process (thus no manual Q override). 
Note there is one exception to this rule: When the data does not contain Q-values (you can think of this data as corrupt), then the user can manually insert them.